src/cpp/rtps/writer/StatelessWriter.cpp

```diff
@@ -407,7 +407,7 @@ bool StatelessWriter::wait_for_acknowledgement(
     uint64_t sequence_number = seq.to64long();
     auto change_is_acknowledged = [this, sequence_number]()
             {
-                return sequence_number >= last_sequence_number_sent_;
+                return sequence_number <= last_sequence_number_sent_;
```

Comment:
We should also return `true` here if the sequence number is not present in the history.